src/irr/asset/CGLIWriter.cpp

```diff
@@ -153,12 +153,28 @@ namespace irr
 			{
 				const auto ptrBeginningOfRegion = data + region->bufferOffset;
 				const auto layerSize = getFullSizeOfLayer(region->imageSubresource.mipLevel);
-
+				const auto textureGliImgHeight = texture.extent(region->imageSubresource.mipLevel).y;
+				const auto textureGliStride = texture.extent(region->imageSubresource.mipLevel).x;
+				const auto textureGliStrideInBytes = textureGliStride * sizeof(singleChannelByteSize);
+				const auto imgBufferWidth = region->bufferRowLength > 0 ? region->bufferRowLength : region->imageExtent.width;
+				const auto imgBufferWidthInBytes = imgBufferWidth * sizeof(singleChannelByteSize);
```

Comment:
ekhm, what are you expecting to get as an answer from `sizeof(singleChannelByteSize)` ?